jaeger-core/src/main/java/io/jaegertracing/internal/JaegerTracer.java

```diff
@@ -130,13 +129,12 @@ protected JaegerTracer(JaegerTracer.Builder builder) {
       log.info("No shutdown hook registered: Please call close() manually on application shutdown.");
     } else {
       // register this tracer with a shutdown hook, to flush the spans before the VM shuts down
-      shutdownHook = new Thread() {
+      Runtime.getRuntime().addShutdownHook(new Thread() {
         @Override
         public void run() {
           JaegerTracer.this.close();
```

Comment:
I think the proper fix is to set `shutdownHook = null` before calling close().

Reply:
If I understand your suggestion that wouldn't help.

The problem is that when `JaegerTracer#close()` is triggered by shutdown hook (i.e. shutdown is already in progress), then calling `Runtime.removeShutdownHook(shutdownHook)` will fail with IllegalStateException. I.e. we cannot remove hooks when shutdown is already in progress.

Hm, or I could just catch the IllegalStateException, I don't know why I didn't thought of that before :/.

Reply:
I will prepare another PR where the exception is caught.

Reply:
I think it _will_ help. The sequence would be:
  * System.exit()
  * shutdown hook invoked
  * before closing the tracer, we reset `shutdownHook = null`
  * tracer.Close() won't attempt to remove the hook

Reply:
OK, you are right. I'm gonna update currently opened PRs.